source/Octopus.TestPortForwarder/SocketPump.cs

```diff
@@ -48,8 +71,7 @@ public async Task<SocketStatus> PumpBytes(Socket readFrom, Socket writeTo, Cance
                 dataTransferObserver.WritingData(tcpPump, buffer);
 
                 await PausePump(cancellationToken);
-
-                await WriteToSocket(writeTo, buffer.GetBuffer(), (int)buffer.Length, cancellationToken);
+                await WriteToSocketDelayingSendingTheLastNBytes(writeTo, buffer.GetBuffer(), (int)buffer.Length, numberOfBytesToDelaySending, cancellationToken);
```

Comment:
Why pass this in when it's a member variable?

Reply:
It used to be a static, which I have changed it back to. For the purposes of functional is a little easier to reason about.

Reply:
I personally find it more confusing having it there. But it's not a big deal.